modules/metrics/README.md

```diff
@@ -157,6 +157,7 @@ will then emit a '0' value until the first time a certain status code shows up.
 key | type | required | note
 --- | ---- | -------- | ----
 `ffwd.interval` | int | optional | interval in seconds of reporting metrics to ffwd; default 30
+`ffwd.flush` | boolean | optional | include a final flush of metrics on service shut down; default `False`
```

Comment:
Is there a reason to make this configurable as flushing metrics on shutdown seems like the behaviour we always want?

Reply:
The idea was not to force new behavior on users that don't have a problem with the way it was working previously. 
We could make it the default though.

Reply:
Changed this to default to "True". Users can still opt out to the previous functionality if they'd like with the flag set to "False". Tt adds a block to shutdown so some people might care.

Reply:
Good point regarding blocking shutdown. 
I guess that is only relevant for the HttpReporter as the other uses UDP? 
How long will the HttpReporter block if the metrics api would be unavailable ?

Reply:
Internally, there are very few  (if any) people using the http reporter. We've been debating deprecating it and removing it. It doesn't block if metricsapi is unavailable. It just takes a bit longer for shutdown since it's trying to run that final flush of metrics, as opposed to previously it would immediately send shutdown and nothing else. 

The remaining issue is that ffwd would need to wait a bit to really get *all* the metrics, but since it's part of the same pod it will get a sigterm when the main application does. A more guaranteed flush of all the metrics would probably require some kind of wait or timeout.